### DISCUSSION TITLE
Can you just create an api to display only dynamic album art ?

### DISCUSSION THREAD

**Discussion by math14is** [AzuraCast/AzuraCast#5690] [General]
Can you just create an api to display only dynamic album art ?

**Comment by Mozitux** [comment]
Hello, can you explain a little more exactly what you are trying to do?

**Comment by math14is** [comment]
I would like a link that returns the covers. and only that

**Reply by Mozitux** [reply]
Links to album art are linked to each track, and since each track can have different cover art, the link will be different for each new music that plays on the station.

On the other hand, if you explain to us how you want to integrate the cover images, we may be able to help you.

**Reply by Mozitux** [reply]
If you want to, here is an example of a link leading to a music cover. But as I said above, the cover will not change over the titles.
https://{domain_name}/api/station/{station_number}/art/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.jpg

**Comment by math14is** [comment]
that's the concern. Thank you for your reply

**Comment by BusterNeece** [comment]
The rolling release now has an endpoint specifically to show the current artwork for a given station:

```
/api/nowplaying/(station_id)/art
```

Or, if you want to include a UNIX timestamp and file extension for cache busting purposes:

```
/api/nowplaying/(station_id)/art/(timestamp).jpg
```

Where `station_id` is the station's numeric ID or short_name, exactly as is used in the main NowPlaying APIs.

We've already started using this as the open-graph image on public pages, so if you embed the public page into a social media post, the current playing album art should appear.

**Comment by math14is** [comment]
very good, thank for you reponse

Le jeu. 10 nov. 2022 à 01:18, Buster Neece ***@***.***> a
écrit :

**Comment by Moonbase59** [comment]
Thanks for showing us the cache busting trick!

**Comment by math14is** [comment]
?

**Reply by Moonbase59** [reply]
Was meant for @BusterNeece.

@math14is: "Cache busting" is a trick to avoid browsers caching stuff that shouldn’t be cached. Typically, you’d add some unique id or timestamp to something so the browser thinks it’s new content and loads it, instead of using the cache.

@BusterNeece showed us that this "cache busting trick" actually also works when using the API!

**Comment by walliesdevelopments** [comment]
Add this into your custom JS and it will do the public page am yet to see if i can edit the Azuracast Metadata from here 

```
// Wrap the entire code in a DOMContentLoaded event listener
document.addEventListener('DOMContentLoaded', function () {
  // Define the myFunction function
  function myFunction() {
    // Your code here
    const apiUrl = 'https://tools.elevatehosting.co.uk/api/v2/lookup/song';
    const nowPlayingMainDiv = document.querySelector('.now-playing-main');

    // Check if the nowPlayingMainDiv exists before proceeding
    if (nowPlayingMainDiv) {
      const artistElement = nowPlayingMainDiv.querySelector('.now-playing-artist');
      const titleElement = nowPlayingMainDiv.querySelector('.now-playing-title');

      // Check if the elements exist before accessing their text content
      if (artistElement && titleElement) {
        const artist = artistElement.textContent;
        const title = titleElement.textContent;
        console.log('Artist:', artist);
        console.log('Title:', title);

        // Construct the search URL with artist and title parameters
        const searchURL = apiUrl + '?artist=' + artist + '&title=' + title;

        // Make a GET request to the API
        fetch(searchURL)
          .then(response => {
            // Check if the response status is OK (status code 200)
            if (!response.ok) {
              throw new Error('Network response was not ok');
            }
            // Parse the response as JSON
            return response.json();
          })
          .then(data => {
            // Use the fetched data
            console.log(data.result.covers.big);

          // Get a reference to the div element
var div = document.querySelector('.now-playing-art');

// Find the anchor tag within the div
var anchor = div.querySelector('a');

// Find the image tag within the anchor tag
var image = anchor.querySelector('img');
// Change the image source
image.src = data.result.covers.big;

// Optionally, update the alt text
image.alt = 'Error';

// Optionally, update the href attribute of the anchor tag if necessary
            var hrefcombine = 'https://open.spotify.com/track/' + data.result.spotify_id
anchor.href = hrefcombine;

          })
          .catch(error => {
            // Handle any errors that occurred during the fetch
            console.error('Fetch error:', error);
          });

        // Your code here
        console.log('This code runs every 10 seconds.');
      } else {
        console.error('One or more elements not found.');
      }
    }
  }

  // Call myFunction initially
  myFunction();

  // Then, call myFunction every 10 seconds
  setInterval(myFunction, 10000);
});

```
